src/security/builtin_plugins/authentication/CMakeLists.txt

```diff
@@ -0,0 +1,65 @@
+#
+# Copyright(c) 2006 to 2019 ADLINK Technology Limited and others
+#
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License v. 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0, or the Eclipse Distribution License
+# v. 1.0 which is available at
+# http://www.eclipse.org/org/documents/edl-v10.php.
+#
+# SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+#
+PREPEND(srcs_authentication "${CMAKE_CURRENT_LIST_DIR}/src"
+    authentication.c
+    auth_utils.c
+)
+ 
+PREPEND(hdrs_private_authentication "${CMAKE_CURRENT_LIST_DIR}/src"
+    auth_defs.h
+    authentication.h
+    auth_utils.hdds_security_core_if.h
+)
+
+
+add_library(dds_security_auth SHARED "")
+
+find_package(OpenSSL )
+
+target_link_libraries(dds_security_auth PUBLIC ddsc)
+target_link_libraries(dds_security_auth PUBLIC OpenSSL::SSL)
+
+add_definitions(-Dddsc_EXPORTS)
```

Comment:
Is this necessary?

Reply:
Actually, I was using this definition to say that we are exporting authentication plugin because I use DDS_EXPORT before the functions and DDS_EXPORT is defined in WINDOWS generated export.h as follows:
```
#  ifndef DDS_EXPORT
#    ifdef ddsc_EXPORTS
        /* We are building this library */
#      define DDS_EXPORT __declspec(dllexport)
#    else
        /* We are using this library */
#      define DDS_EXPORT __declspec(dllimport)
#    endif
#  endif
```
So I had to define ddsc_EXPORTS to get export symbol in Windows

Reply:
It doesn't work like that: this way, if you include one of the include files declaring symbols in the "ddsc" library, it will treat them as exported instead of as imported, in any case you get a mess this way.

Each shared library needs its own define — here you need to end up with an exports.h file for the plugin, with the corresponding defines.